scripts/list_init_hooks.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python
+#
+# Scripts for print all init hooks in f9-kernel.
+# Copyright (c) 2016 The F9 Microkernel Project. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# Description:
+#       At now this script will print by hook level,
+#       but is not the real seq when kernel perform run_init_hook.
+#
+
+# ----------------- LIBRARIES --------------------
+
+import os
+import re
+import sys
+import subprocess
+from collections import OrderedDict
+
+# ------------------- CONST --------------------
+
+# Define in kernel/init_hook.h
+HOOK_LEVEL = OrderedDict([])
+HOOK_LEVEL.update({'INIT_LEVEL_EARLIEST': 1})
+HOOK_LEVEL.update({'INIT_LEVEL_PLATFORM_EARLY': 0x1000})
+HOOK_LEVEL.update({'INIT_LEVEL_PLATFORM': 0x2000})
+HOOK_LEVEL.update({'INIT_LEVEL_KERNEL_EARLY': 0x3000})
+HOOK_LEVEL.update({'INIT_LEVEL_KERNEL': 0x4000})
+HOOK_LEVEL.update({'INIT_LEVEL_LAST': 0xFFFFFFFF})
+
+# Dynamiclly adding variable on the fly
+for hook in HOOK_LEVEL:
+    exec('%s = %d' % (hook, HOOK_LEVEL[hook]))
+
+
+# ----------------- CLASS --------------------
+class MakefileConfig(dict):
+    __getattr__ = dict.__getitem__
+    __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def __init__(self, infile=None):
+        # Hack for board build prefix
+        self.board = {
```

Comment:
Is about `self.board` or other thing ?

The problem here is we need to specify user's config board for build/prefix, so only read from `.config` file and program can determine which board is using now.

And pair of board -> build prefix was existed in top dir `Makefile` if else block
```
ifeq "$(CONFIG_BOARD_STM32F429DISCOVERY)" "y"
BOARD ?= discoveryf429
else ifeq "$(CONFIG_BOARD_STM32P103)" "y"
BOARD ?= stm32p103
else
BOARD ?= discoveryf4
endif
```
rather than making a turly parser, I make this attr set when parsing `.config`, since we will only get one CONFIG_BOARD=y in config file.